piet-gpu/shader/kernel4.comp

```diff
@@ -39,6 +41,27 @@ layout(set = 0, binding = 3) uniform sampler2D textures[];
 
 shared MallocResult sh_clip_alloc;
 
+// The lookup table is implemented based on the description of Efficient GPU Path Rendering Using Scanline Rasterization,
+// Li et al. 2016. The coordinate mapping could potentially use some improvement (since the paper does not mention how
+// they derived it), but for now it's implemented verbatim from the paper's description.
+// The lookup table is precomputed in piet_gpu::Renderer::make_winding_lut.
+uint getLut(vec2 n, float c) {
+    uint mask = 0;
+    if (c < 0) {
+        n = -n;
+        c = -c;
+        mask = ~0;
+    }
+    if (c >= 1.) {
+        // mask ^= 0;
+    } else {
+        vec2 tex_coord = (0.5 - 0.5 * c) * n + vec2(0.5, 0.5);
+        uvec4 tex = texture(winding_lut, tex_coord);
```

Comment:
Can this be replaced with imageLoads?

Texture sampling from compute programs is relatively slow on lower end hardware, probably because of the unpredictable sampling coordinates. Also, compiling kernel4 for CPUs through something like SwiftShader is much harder with texture sampling. See my https://github.com/linebender/piet-gpu/pull/63/ that replaces texture(Grad) with imageLoad.

Reply:
Yeah, it surely makes sense to replace this with an image. Though due to the piet-gpu-hal mess this would mean partially applying #63 which I don't want to do right now. I'll probably make the switch when it's easy enough to do.